### DISCUSSION TITLE
Determine variable names of query

### DISCUSSION THREAD

**Discussion by jaulz** [urql-graphql/urql#3175] [Q&A] [answered]
Is there any possibility to determine the variable names of a query? It would be very handy to determine if certain variables are available (e.g. `page`) to let the UI display corresponding controls.

**Comment by JoviDeCroock** [comment]
Not sure what you mean but variables live entirely on your side

**Reply by jaulz** [reply]
@JoviDeCroock yeah, but for example I would like to create a helper function that automatically derives whether the query is "paginatable"(?) or not without explicity defining it so I was actually looking for the parsed query document and hoped there was something like a `document` variable:
```typescript
const query = useQuery({ query: 'query ListTeams($page: Int) { teams(page: $page) }')
const paginatable = !!query.document.variables.page
```

**Reply by JoviDeCroock** [reply, marked as answer]
You can use res.operation.variables or look at the query-ast to find variableDefinitions

**Reply by jaulz** [reply]
Ahh, sorry, I didn't see the exposed `query` on `operation`. Thanks a lot and have a nice weekend! 😊
